test/suite/test_cursor_bound17.py

```diff
@@ -0,0 +1,105 @@
+        ('bytes', dict(key_format='u')),
+        ('composite_string', dict(key_format='SSS')),
+        ('composite_int_string', dict(key_format='iS')),
+        ('composite_complex', dict(key_format='iSru')),
+    ]
+
+    value_formats = [
+        ('string', dict(value_format='S')),
+        ('complex-string', dict(value_format='SS')),
+    ]
+
+    config = [
+        ('inclusive-evict', dict(lower_inclusive=True,upper_inclusive=True,evict=True)),
+        ('no-inclusive-evict', dict(lower_inclusive=False,upper_inclusive=False,evict=True))
+    ]
+
+    direction = [
+        ('prev', dict(next=False)),
+        ('next', dict(next=True)),
+    ]
+
+    scenarios = make_scenarios(types, key_formats, value_formats, config, direction)
+
+    def test_bound_api(self):
+        uri = self.uri + self.file_name
+        create_params = 'value_format={},key_format={}'.format(self.value_format, self.key_format)
+        if self.use_colgroup:
+            create_params += self.gen_colgroup_create_param()
+        self.session.create(uri, create_params)
+
+        # Add in column groups.
+        if self.use_colgroup:
+            for i in range(0, len(self.value_format)):
+                create_params = 'columns=(v{0}),'.format(i)
+                suburi = 'colgroup:{0}:g{1}'.format(self.file_name, i)
+                self.session.create(suburi, create_params)
+
+        cursor = self.create_session_and_cursor()                                                                                                                                                                                                                                        
+
+        # Test default bound api with column groups.
+        self.assertEqual(self.set_bounds(cursor, 40, "lower"), 0) 
+        self.assertEqual(self.set_bounds(cursor, 90, "upper"), 0)
+
+        # Test that original bounds persist even if setting one bound fails. 
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: self.set_bounds(cursor, 30, "upper"), '/Invalid argument/')
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: self.set_bounds(cursor, 95, "lower"), '/Invalid argument/')
+
+        self.assertEqual(self.set_bounds(cursor, 50, "lower"), 0) 
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: self.set_bounds(cursor, 40, "upper"), '/Invalid argument/')
+        self.cursor_traversal_bound(cursor, 50, 90, None)
+
+        self.assertEqual(self.set_bounds(cursor, 70, "upper"), 0) 
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: self.set_bounds(cursor, 80, "lower"), '/Invalid argument/')
+        self.cursor_traversal_bound(cursor, 50, 70, None)
+
+        #Test successful setting of both bounds.
+        self.assertEqual(self.set_bounds(cursor, 50, "lower"), 0) 
+        self.assertEqual(self.set_bounds(cursor, 70, "upper"), 0) 
+        self.cursor_traversal_bound(cursor, 50, 70, None)
+        
```

Comment:
np: new line at end of file

Reply:
[5b20368](https://github.com/wiredtiger/wiredtiger/pull/8265/commits/5b203680c47ab284e59883d194495a9a6703223e)